pkg/apis/eksctl.io/v1alpha5/types.go

```diff
@@ -666,6 +668,14 @@ type ClusterProvider interface {
 	WaitTimeout() time.Duration
 	ConfigProvider() client.ConfigProvider
 	Session() *session.Session
+
+	ServicesV2() ServicesV2
```

Comment:
It might be easier to have the new interfaces live directly on the struct? Like
```suggestion
	STSV2() stsifacev2.STS
```

So that when we come to remove the orginal STS its just a:
- Remove `STS`
- Rename `STSV2` -> `STS`

The current method means this migration would be a bit more fiddly than that. We would have to:
- Remove `STS`
- Move `STS` from `ServicesV2` to the main interface

Reply:
I tried to avoid that approach because it polluted the `ClusterProvider` interface with more methods and required suffixing all service methods with `V2`, but thinking about it more, it requires more changes, so I'll go ahead with adding V2 methods directly on `ClusterProvider`.